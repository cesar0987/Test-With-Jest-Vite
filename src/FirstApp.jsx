import PropTypes from 'prop-types';
const FirstApp = ({title,subTitle,name}) => {
    return (
        <>
        <h1>{title}</h1>
        <p>{subTitle}</p>
        <p>{name}</p>
        </>
    );
    }

export default FirstApp;

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}