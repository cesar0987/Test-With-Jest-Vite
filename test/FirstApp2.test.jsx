import { render } from '@testing-library/react';
import FirstApp from '../src/FirstApp';


describe('Prueba en <FirstApp/>', () => {
    const title = 'Hola soy Goku';
    test('debe hacer match con el snapshot', () => {
        const {container} = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();
    })

    test('Debe mostrar el mensaje"Hola soy Goku"', () => {
        const {getByText} = render(<FirstApp title={title}/>);
        expect(getByText(title)).toBeInTheDocument();
    }
    )
    
})