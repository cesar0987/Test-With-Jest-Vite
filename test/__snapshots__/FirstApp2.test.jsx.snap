// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <FirstApp/> debe hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola soy Fernando
  </div>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Cesar Fernando
  </p>
</div>
`;
